### DISCUSSION TITLE
Maven registry authentication failed with any token on GitHub Packages

### DISCUSSION THREAD

**Discussion by FelipeMenezesDM** [community/community#49836] [Packages]
### Select Topic Area

Bug

### Body

I'm trying to install maven dependencies from GitHub Packages, but the registry is not allowing authentication.
Deploying packages works normally, but installing them in another application does not.
Despite being in public repositories, authentication fails and the message below is displayed when I run `mvn install`:

```
Downloading from spring-secret-manager-suite: https://maven.pkg.github.com/felipemenezesdm/spring-secret-manager-suite/br/com/felipemenezesdm/spring-secret-manager-suite/1.0.0/spring-secret-manager-suite-1.0.0.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.187 s
[INFO] Finished at: 2023-03-11T23:35:26-03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project auth-server-connect-spring: Could not resolve dependencies for project br.com.felipemenezesdm:auth-server-connect-spring:jar:1.0.0: Failed to collect dependencies at br.com.felipemenezesdm:spring-secret-manager-suite:jar:1.0.0: Failed to read artifact descriptor for br.com.felipemenezesdm:spring-secret-manager-suite:jar:1.0.0: Could not transfer artifact br.com.felipemenezesdm:spring-secret-manager-suite:pom:1.0.0 from/to spring-secret-manager-suite (https://maven.pkg.github.com/felipemenezesdm/spring-secret-manager-suite): authentication failed for https://maven.pkg.github.com/felipemenezesdm/spring-secret-manager-suite/br/com/felipemenezesdm/spring-secret-manager-suite/1.0.0/spring-secret-manager-suite-1.0.0.pom, status: 401 Unauthorized -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException

Process finished with exit code 1
```

Details
- Dependency configuration:
  ```
  ## pom.xml
  <dependency>
            <groupId>br.com.felipemenezesdm</groupId>
            <artifactId>spring-secret-manager-suite</artifactId>
            <version>1.0.0</version>
        </dependency>

  ## settings.xml
  <activeProfiles>
        <activeProfile>general</activeProfile>
    </activeProfiles>

    <profiles>
        <profile>
            <id>general</id>
            <repositories>
                <repository>
                    <id>central</id>
                    <url>https://repo1.maven.org/maven2</url>
                </repository>
                <repository>
                    <id>spring-secret-manager-suite</id>
                    <url>https://maven.pkg.github.com/felipemenezesdm/spring-secret-manager-suite</url>
                    <snapshots>
                        <enabled>true</enabled>
                    </snapshots>
                </repository>
            </repositories>
        </profile>
    </profiles>
  ```
- The error is displayed both locally and via GitHub Actions.
- I've tried generating a new personal token with all permissions and also using $GITHUB_TOKEN in github actions, but both didn't work.
- I already granted permission to read and write packages to $GITHUB_TOKEN.
- I already configured a server in settings.xml
  ```
  <servers>
        <server>
            <id>github</id>
            <username>${repo.usrnm}</username>
            <password>${repo.pswd}</password>
        </server>
    </servers>
  ```
- This worked perfectly a few days ago

Repo: https://github.com/FelipeMenezesDM/auth-server-connect-spring

**Comment by bbkx226** [comment]


**Reply by FelipeMenezesDM** [reply]
Hello, @bbkx226. Thanks for you answer.
As I mentioned in the comment, I already checked access permissions (even if the packages are public) and even generated other personal tokens, but the problem persists.
As I also explained, this was working a few days ago, but today I just updated the README of the package and the problem appeared.

**Reply by bbkx226** [reply]
My bad, I didn't catch what you said earlier. If that's really the deal, I don't have a clue how to fix it. Maybe we gotta wait for the others

**Comment by manuelmsperanza** [comment]
I'm facing the same issue (_Unauthorized (401)_) when I do _mvn release:perform_:

[INFO] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-deploy-plugin:3.0.0:deploy (default-deploy) on project loggingcorner: Failed to deploy artifacts: Could not transfer artifact com.hoffnungland:loggingcorner:pom:0.0.26 from/to manuelmsperanza.github (https://maven.pkg.github.com/manuelmsperanza/loggingcorner): status code: 401, reason phrase: Unauthorized (401) -> [Help 1]

but it's working when I do mvn deploy for the *-SNAPSHOT.

I thought it was matter of jdk version, but not.

**Comment by manuelmsperanza** [comment]
I resolved adding the Autorization Beares, as per [mvn deploy in Github actions gives 401 Unauthorized](https://stackoverflow.com/questions/61864370/mvn-deploy-in-github-actions-gives-401-unauthorized)

> For starters, you will need a settings.xml file. Here is my minimal setup:


	<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
		  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
		  xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
							  https://maven.apache.org/xsd/settings-1.0.0.xsd">
	  <servers>
		<server>
		  <id>github</id>
		  <configuration>
			<httpHeaders>
			  <property>
				<name>Authorization</name>
				<value>Bearer <GITHUB_TOKEN_GOES HERE></value>
			  </property>
			</httpHeaders>
		  </configuration>
		</server>
	  </servers>
	</settings>

HTH

**Reply by hongchanphat** [reply]
@manuelmsperanza thanks, you save my day after many hours of stuck.
